server/internal/server/pod.go

```diff
@@ -0,0 +1,75 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/llm-operator/inference-manager/pkg/k8s"
```

Comment:
Slightly prefer moving this package under `server/internal/k8s` as `engine` doesn't use. We can create a separate PR.